### PR TITLE
Adjust integrationtests after #127

### PR DESCRIPTION
**Pull request by TobiasNx**
Since Fix can create array with `$append` since #127. I thought we could simplify our integrationtests and delete all `set_arrays` that are only used to create empty arrays.

As result two integration tests seem to fail now.

1) This seems to be an undetected bug, `replace_all` now copies an element it should not:

https://github.com/metafacture/metafacture-fix/blob/d18a5035d2b74508d685ec863c91e4382aff2c26/metafix/src/test/resources/org/metafacture/metafix/integration/method/fromXml/toJson/replace_allInOptionalSubfieldOfRepeatedObjectsWithAsterisk/test.fix

I opened a new issue for this: #377

2) https://github.com/metafacture/metafacture-fix/blob/d18a5035d2b74508d685ec863c91e4382aff2c26/metafix/src/test/resources/org/metafacture/metafix/integration/record/fromJson/toJson/copy_fieldToArrayOfStringsWithIndex/test.fix

Catmandu creates new arrays with index numbers.

```
$ echo '{}' | catmandu convert JSON to YAML --fix 'add_field(list.0.b,test)'
---
list:
- b: test
...

$ echo '{}' | catmandu convert JSON to YAML --fix 'add_field(list.3.b,test)'
---
list:
- ~
- ~
- ~
- b: test
...

```

I am not sure if I am in favour of this behaviour since this blocks the possibilty of objects with numeric names.

### CODE REVIEW

**Review by blackwinter — CHANGES_REQUESTED** [review]
I would prefer to leave the old integration test for `method/fromXml/toJson/replace_allInOptionalSubfieldOfRepeatedObjectsWithAsterisk` untouched and instead add a new one for #377.

**Review by fsteeg — COMMENTED** [review]
Agree about adding a new test, otherwise +1

**Comment by TobiasNx** [comment]
Done

**Review by fsteeg — COMMENTED** [review]
There's a typo in the path to the new test (Asteris -> Asterisk).

**Comment by TobiasNx** [comment]
Typo fixed

**Review by blackwinter — APPROVED** [review]
The new test name could reuse the convention from the unit tests (`ImplicitArray` instead of `WithoutCreatingArrayWithSetArray`).

**Review by fsteeg — APPROVED** [review]

File: metafix/src/test/resources/org/metafacture/metafix/integration/method/fromXml/toJson/replace_allInOptionalSubfieldOfRepeatedObjectsWithAsteriskWithoutCreatingArrayWithSetArray/todo.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+See #377 . This behaviour seems to be undetected or new after #369
```
[the remark this answers is not in the document]
Comment (TobiasNx) [reply metafix/src/test/resources/org/metafacture/metafix/integration/method/fromXml/toJson/replace_allInOptionalSubfieldOfRepeatedObjectsWithAsteriskWithoutCreatingArrayWithSetArray/todo.txt]: Adjusted as suggested. Also renamed the file as suggested.

**Comment by TobiasNx** [comment]
I changed a set_array specific integration test, I need to change it back: metafix/src/test/resources/org/metafacture/metafix/integration/record/fromJson/toJson/set_arrayWithoutValuesAndMove_fieldNewArrayOfStringsIntoArrayOfObjects/test.fix

**Pull request merged**